core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java

```diff
@@ -177,14 +192,72 @@ public synchronized void downgradeLastWithScheme(LockingScheme lockingScheme) {
     }
   }
 
+  /**
+   * Returns the closest ancestor of the target inode (last inode in the full path).
+   *
+   * @return the closest ancestor inode
+   * @throws FileDoesNotExistException if an ancestor does not exist
+   */
+  public synchronized Inode<?> getAncestorInode() throws FileDoesNotExistException {
+    int ancestorIndex = mPathComponents.length - 2;
+    if (ancestorIndex < 0) {
+      throw new FileDoesNotExistException(ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(mUri));
+    }
+    ancestorIndex = Math.min(ancestorIndex, mLockList.getInodes().size() - 1);
+    return mLockList.getInodes().get(ancestorIndex);
+  }
+
+  /**
+   * Constructs a temporary {@link LockedInodePath} from an existing {@link LockedInodePath}, for
+   * a direct child of the existing path. A new {@link LockedInodePath} object is returned. When
+   * the returned temporary path is closed, it does not close the existing path.
+   *
+   * @param childName the name of the direct child
+   * @return a {@link LockedInodePath} for the direct child
+   * @throws InvalidPathException if the path is invalid
+   */
+  public synchronized LockedInodePath createTempPathForChild(String childName)
+      throws InvalidPathException {
+    Preconditions.checkNotNull(getInodeOrNull());
+    Preconditions.checkState(getInodeOrNull().isDirectory(),
+        "Trying to create TempPathForChild for a file inode");
+    return new MutableLockedInodePath(mUri.join(childName), new CompositeInodeLockList(mLockList),
+        mLockMode);
+  }
+
+  /**
+   * Constructs a temporary {@link LockedInodePath} from an existing {@link LockedInodePath}, for
```

Comment:
The comments should distinguish between `createTempPathForChild` and `createTempPathForExistingChild`. I think this one is for an existing inode, and the other one just adds to the path, so the inode is not locked yet (could not exist).

Reply:
addressed